### DISCUSSION TITLE
Sharing Parent Atom's value in the Chidlren Atoms

### DISCUSSION THREAD

**Discussion by wootra** [pmndrs/jotai#1577] [Ideas] [answered]
Hi, team. 
Firstly, thanks for the great library which makes everyone's life so easy.
Recently, we were encountering a problem while taking advantage of Jotai.Provider.  
As you know, we can create multiple provider like React.Context, but works pretty differently.
```
<Jotai.Provider scope="parent-scope"> // in the parent component
  <Jotai.Provider scope="child-scope"> // in a child component
   ...
  </Jotai.Provider>
</Jotai.Provider>
```
roughly, the structure is like above. 
but we cannot use any atom value that is used in the parent-scope from the atom that is used in chid-scope. 
I know it is not something Jotai is supporting - sharing atom values crossing scope boundary. But there is a case that we should share the value while parent component is still alive.
We already used custom event to control the atom in the parent boundary. but taking a value from it is another problem because if you use get(atomInParent) will just return `undefined` without any warning. 

I am trying to solve this problem making a custom provider like this:
```
const sharedObject = {};
const setSharedValue = (scopeName, variableName, variableValue)=>{
  if(sharedObject[scopeName]) {
    sharedObject[scopeName][variableName] = variableValue;
  }
  else throw new Error("sharedObject["+scopeName+"]["+variableName+"] does not exist");
}
const getSharedValue = (scopeName, variableName)=>{
  if(!sharedObject[scopeName]?.[variableName]) throw new Error("sharedObject["+scopeName+"]["+variableName+"] does not exist");
  return sharedObject[scopeName]?.[variableName];
}
const JotaiProviderWithSharedObj = ({scopeName, children})=>{
  useEffect(()=>{
    sharedObject[scopeName] = {};
    return ()=>{
      sharedObject[scopeName] = null;
      delete sharedObject[scopeName];
    }
  }
  return <JotaiProvider scope={scopeName}>{children}</JotaiProvider>
}
```

and in some Atom that I want to access, I am registering the value like this:
```
const someParentAtom = atom(get=>{
   ...
    return someThing;
});
const useSomeParent = ()=>{
  const someValueToReturn = useAtomValue(someParentAtom, "someParentAtom");
  setSharedValue("parent-scope", "someParentAtom", valueToReturn);
  return someValueToReturn;
}
```

so we can get the "saved" value in the child scope without making hook/passing the verbose argument.
```
const someChildAtom = atom(get=>{
  const valueInParentAtom = getSharedValue("parent-scope", "someParentAtom");
 ...do some stuff;
 return someNewValue;
});

```

What do you think about this? Or do you guys have an insights that would be better?

**Comment by dai-shi** [comment, marked as answer]
Provider scope has been misguided for the usage. It was meant only for library use.
In Jotai v2, `scope` will be removed. Please avoid using it.

Provider just works like React Context, and it is just a React Context Provider.

I haven't looked into your example closely, but I'm sure what you are looking for is something like [jotai-molecules](https://github.com/saasquatch/jotai-molecules).
